### ISSUE TITLE
Support for HTTP/1.0 Healthchecks by Loadbalancers

### ISSUE THREAD

**Issue by daimoniac** [datawire/ambassador#988]
**Please describe your use case / problem.**
Some loadbalancer vendors are using http/1.0 healthchecks (haproxy default, digitalocean exclusively)
When deploying ambassador behind such a loadbalancer, Ambassador/envoy is replying with 426 UPGRADE REQUIRED answers:
```
ACCESS [2018-11-21T10:12:10.359Z] "GET /ambassador/v0/check_alive HTTP/1.0" 426 - 0 0 0 - "-" "curl/7.58.0" "-" "localhost:8080" "-"
ACCESS [2018-11-21T10:12:18.139Z] "GET /ambassador/v0/check_ready HTTP/1.0" 426 - 0 0 0 - "-" "curl/7.58.0" "-" "localhost:8080" "-"
```
Therefore, the upstream Loadbalancers are marking the service as DOWN (they expect a 200 OK answer).

**Describe the solution you'd like**
I think we should be able to answer simple http/1.0 calls. 
envoy does support it *accept_http_10*
https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/core/protocol.proto

**Describe alternatives you've considered**
I have now resolved it by manually setting the upstream loadbalancer to TCP only check.
So the Loadbalancer only checks if the port ist open. Thats not optimal, but better than no service

**Comment by rljohnsn** [comment]
Seeing the same thing.

```
"GET /public/ping HTTP/1.0" 426 - 0 0 0 - "-" "NewRelicPinger/1.0 (1144731)"
```

**Comment by phenoxim** [comment]
Same issue with me. Could anyone give a hint to resolve this propblem?

**Comment by mstrother** [comment]
I'm also looking for HTTP 1.0 support.

**Comment by phenoxim** [comment]
> Same issue with me. Could anyone give a hint to resolve this propblem?

FYI

Update http_config here:
https://github.com/datawire/ambassador/blob/7ccc4945f7b50af35c77ffddca2cc7f22979882a/ambassador/ambassador/envoy/v2/v2listener.py#L353

```
http_config['http_protocol_options'] = {
        'accept_http_10': True
}
```

and build a private ambassador image.

Solved this issue.

**Comment by ryanmaclean** [comment]
@phenoxim - thanks for the patch! This helps me quite a bit. I've forked the repo, and am currently testing, but will submit a PR once done.

**Comment by richarddli** [comment]
A PR would be awesome. We'll send a t-shirt over :-)

**Comment by kflynn** [comment]
#1484 fixes this. Thanks, @cyrus-mc and @matthewce!

**Comment by richarddli** [comment]
Fixed in 0.61.
